test/make_dist.py

```diff
@@ -135,7 +136,23 @@ def download_dist():
         if len(names) != 1 or not names[0].endswith(".tar.xz"):
             print("make_dist: expected zip artifact with exactly one tar.xz member")
             return None
-        return fzip.extract(names[0])
+        tar_path = fzip.extract(names[0])
+
+    # Extract node_modules and dist locally for speeding up the build and allowing integration tests to run
+    # unfortunately it is impossible with tar CLI and Python's tarfile to just extract that subset;
+    # so extract it all, pick what we want, and clean it up
+    unpack_dirs = [d for d in ["dist", "node_modules"] if not os.path.exists(d)]
```

Comment:
This is not necessary.  After playing around a bunch I was able to get this working with just plain `tar`.  You need to include the directory name of the archive, like `cockpit-222/dist/` and then it should work...

Also: `tar -m` is your friend.

Reply:
I couldn't make it work without already knowing the directory name in advance, as the pattern matching is too inflexible (`*` always matches across `/`). That would mean either to call `tar -tf` first (duplicating the xunzip, no thanks) or deriving it from the file name (which seems a bit brittle).

Reply:
I added `--touch` (I prefer long options in code for easier understanding), but I kept the unpack logic for now. We can discuss a better solution here or on IRC

Reply:
Deriving from the filename is completely legit.  automake literally has a variable called `$(distdir)` and it names the tarball based on the value of this variable.

Reply:
Done, thanks!